plugins/outputs/remotefile/remotefile.go

```diff
@@ -205,15 +220,25 @@ func (f *File) Write(metrics []telegraf.Metric) error {
 				f.Log.Errorf("Could not serialize metrics: %v", err)
 				continue
 			}
-			groupBuffer[fn] = serialized
+			octets, err := f.encoder.Encode(serialized)
+			if err != nil {
+				f.Log.Errorf("Could not compress metrics: %v", err)
+				continue
+			}
+			groupBuffer[fn] = octets
 		} else {
 			for _, m := range fnMetrics {
 				serialized, err := serializer.Serialize(m)
 				if err != nil {
 					f.Log.Debugf("Could not serialize metric: %v", err)
 					continue
 				}
-				groupBuffer[fn] = append(groupBuffer[fn], serialized...)
+				octets, err := f.encoder.Encode(serialized)
+				if err != nil {
+					f.Log.Debugf("Could not compress metric: %v", err)
```

Comment:
Above you are showing this as an error and here as a debug message. Please choose one and use it in both places!

Reply:
Yeah. Good catch. Made Errorf